sdks/java/core/src/main/java/org/apache/beam/sdk/values/PCollectionViews.java

```diff
@@ -529,4 +534,49 @@ public String toString() {
       return Collections.<TupleTag<?>, PValue>singletonMap(tag, pCollection);
     }
   }
+
+  /** Accessor for the value of a side input in the current context. */
+  @Internal
+  public interface SideInputContext {
+    <T> T sideInput(PCollectionView<T> view);
+  }
+
+  private static ThreadLocal<SideInputContext> sideInputContextThreadLocal = new ThreadLocal<>();
```

Comment:
<!--new_thread; commit:2b7b6939227e3e105feee6eddf1d6c943c40815a; resolved:0-->
We found out that thread local is pretty slow compared to a ConcurrentHashMap with the key as the current thread. If your worried about leaking information you can always use a guava cache with weak keys.